File: Back/src/main/java/com/smhrd/entity/Alert.java
package com.smhrd.entity;

import java.sql.Timestamp;
import java.time.LocalDateTime;

import com.fasterxml.jackson.annotation.JsonFormat;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;

@Entity
@Data
@Table(name = "tb_alert")
public class Alert {
	
	@Id
	private Integer alert_idx;

	private String user_email;

	private String alert_msg;
	
	@JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")  // 밀리초와 시간대 포함
	private LocalDateTime alerted_at;

	private String received_yn;

	private String farm_name;

	private String sensor_name;

	@JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")  // 밀리초와 시간대 포함
	private LocalDateTime sensing_at;

	@Override
	public String toString() {
		return "Alert 객체";
	}
}

File: Front/src/api/alert.js
import { api } from "./apiService";

export const getAlert = async () => {
    try {
      const response = await api.post("/alert/get", {
        user_email : sessionStorage.getItem("email")
      });
      return response.data;
    } catch (error) {
      throw new Error('알람 데이터를 불러오는 중 오류가 발생했습니다.');
    }
  };


  export const addAlert = async (farm_name, alert_msg, sensor_name, sensing_at) => {
    try {
      const response = await api.post("/alert/add", {
        farm_name : farm_name,
        user_email : sessionStorage.getItem("email"),
        alert_msg : alert_msg,
        sensor_name : sensor_name,
        sensing_at : sensing_at
      });
      return response.data;
    } catch (error) {
      throw new Error('알람 데이터를 저장하는 중 오류가 발생했습니다.');
    }
  };
